### PR TITLE
LPS-143454 Restric the VulcanBatchEngineTaskItemDelegates generated from REST Builder

### PR DESCRIPTION
**Pull request by javierdearcos**
There were many cases where the API Resources were implementing the VulcanBatchEngineTaskItemDelegate interface in resources that has no batch capabilities.

I've restricted the implementation of this interface just in the cases that make sense, where there are batch endpoints or at least there is an endpoint to return a list of resources that can be used to export the resources.

This way we get rid of a lot of false positives that were present mostly in Commerce APIs.

/cc @matijapetanjek Please take a look at the resources that no longer implement VulcanBatchEngineTaskItemDelegate to double-check it makes sense to you (the list of files in the Build REST commit)

### CODE REVIEW

**Comment by javierdearcos** [comment]
ci:forward

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl [hunk 1]
```diff
@@ -609,11 +609,13 @@ public abstract class Base${schemaName}ResourceImpl
...
 	}
 
-	<#if generateBatch>
+	<#if configYAML.generateBatch>
 		public void setVulcanBatchEngineImportTaskResource(VulcanBatchEngineImportTaskResource vulcanBatchEngineImportTaskResource) {
 			this.vulcanBatchEngineImportTaskResource = vulcanBatchEngineImportTaskResource;
 		}
+	</#if>
 
+	<#if generateBatch>
```
Comment (ccorreagg) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl:618]: I find weird to use different parameters for validate batch generations, but I guess in the following PRs we will centralize the functionality to a method in the FreeMarkerTool

Comment (javierdearcos) [reply modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl:618]: As I was in a hurry and I want to forward this PR, so you can rebase yours and send it, I thought to do it this way and centralize in the next PR

**Comment by javierdearcos** [comment]
Seems stuck... 😞

**Comment by javierdearcos** [comment]
ci:stop

**Review by ccorreagg — APPROVED** [review]

**Comment by javierdearcos** [comment]
Sent here https://github.com/liferay-frontend/liferay-portal/pull/2203

**Pull request closed without merging**